Assets/Scripts/Framework/TriggerArea/NetworkTriggerArea.cs

```diff
@@ -0,0 +1,63 @@
+using FishNet.Object;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DerailedDeliveries.Framework.TriggerArea
+{
+    /// <summary>
+    /// A class responsible for handling colliders entering and exiting a trigger area.
+    /// </summary>
+    [RequireComponent(typeof(Collider))]
+    public class NetworkTriggerArea<T> : NetworkBehaviour where T : Component
+    {
```

Comment:
```cs
        private readonly List<T> _componentsInCollider = new();

        /// <summary>
        /// Invoked when a collider enters or exits the trigger area.
        /// </summary>
        public Action<T[]> OnComponentChange;

        /// <summary>
        /// Invoked when a collider enters the trigger area.
        /// </summary>
        public Action<T> OnComponentEnter;

        /// <summary>
        /// Invoked when a collider exits the trigger area.
        /// </summary>
        public Action<T> OnComponentExit;

        /// <summary>
        /// Gets the colliders currently inside the trigger area.
        /// </summary>
        public T[] ComponentsInCollider => _componentsInCollider.ToArray();
```
I was a bit confused by these names, looking at it now I get it but I think it would be more clear if these were named something like OnColliderChange, OnColliderEnter ect.